scripts/api_diff_report/pr_commenter.py

```diff
@@ -0,0 +1,214 @@
+# -*- coding: utf-8 -*-
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import json
+import logging
+import requests
+import argparse
+import api_diff_report
+import datetime
+import pytz
+
+from requests.adapters import HTTPAdapter
+from requests.packages.urllib3.util.retry import Retry
+
+STAGES_PROGRESS = "progress"
+STAGES_END = "end"
+
+TITLE_PROGESS = "## ⏳&nbsp; Detecting API diff in progress...\n"
```

Comment:
I see that this job succeeded in 28 seconds ([log](https://github.com/firebase/firebase-ios-sdk/actions/runs/4898373237/jobs/8747410536?pr=11236)). Do we need a separate "in progress" state?

I think it makes sense for Games SDK, only because their integration tests take several hours to finish.

Reply:
This is because there is not diff in this case. Otherwise, it takes ~5 mins/module.